Config param(s) tests

Rod1122 opened the pull request:
Modified min/max of a value param to be a tuple instead of two separate args.
Unit tests for each derived param class to check validate_value.
Unit test for the parameters container class.

Commit:
Config param(s) tests

Modified min/max of a value param to be a tuple instead of two separate args.
Unit tests for each derived param class to check validate_value.
Unit test for the parameters container class.

TheMinefreak23 reviewed (changes requested):
Pylint has methods to assert that certain exceptions are raised, consider using these instead of asserting a bool you manually set to True on an exception.

https://docs.pytest.org/en/6.2.x/assert.html#assertions-about-expected-exceptions
https://docs.pytest.org/en/6.2.x/reference.html#pytest-raises

Commit:
Modified ConfigParameters test

- reusable add param function with assertions
- using pytest raises functionality
- fixed some linter issues
